plans/example-js/pingpong.js

```diff
@@ -0,0 +1,186 @@
+const os = require('os')
+const net = require('net')
+const ipaddr = require('ipaddr.js')
+const { performance } = require('perf_hooks')
+const { sync, network } = require('@testground/sdk')
+
+async function pingpong (runenv) {
+  runenv.recordMessage('before sync.newBoundClient')
+
+  if (!runenv.testSidecar) {
+    return
+  }
+
+  const client = await sync.newBoundClient(runenv)
+  let server, socket
+
+  try {
+    const netclient = network.newClient(client, runenv)
+
+    runenv.recordMessage('before netclient.waitNetworkInitialized')
+    await netclient.waitNetworkInitialized()
+
+    const oldAddrs = os.networkInterfaces()
+
+    const config = {
+      network: 'default',
+      enable: true,
+      default: {
+        latency: 100 * 1000 * 1000, // 100ms in nanoseconds
+        bandwidth: 1 << 20 // 1 Mib
+      },
+      callbackState: 'network-configured', // Fix this small and big letters thingies.
+      routingPolicy: network.DENY_ALL
+    }
+
+    runenv.recordMessage('before netclient.configureNetwork')
+    await netclient.configureNetwork(config)
+
+    const seq = await client.signalAndWait('ip-allocation', runenv.testInstanceCount)
+
+    // Make sure that the IP addresses don't change unless we request it.
+    const newAddrs = os.networkInterfaces()
+    if (!sameAddrs(oldAddrs, newAddrs)) {
+      throw new Error('interfaces changed')
+    }
+
+    runenv.recordMessage(`I am ${seq}`)
+
+    const ipC = (seq >> 8) + 1
```

Comment:
Considering that `ping-pong` is always using 2 instances (and doesn't work with more than 2 instances), this is a really weird way to update the C and D parts of the IP. I know this is copying the network/ping-pong Go impl., just saying we might want to refactor this soon :)